app/services/child_profiles_services/child_profiles_listing_service.rb

```diff
@@ -77,5 +37,53 @@ def filter_profiles(profiles, filter_type)
         profiles.where('child_reviews.id IS NULL')
       end
     end
+
+    def sort_by_name(children)
+      case @sort_order
+      when Constants::USER_SORTINGS_ORDERS[:asc]
+        children_to_render = filter_profiles(children, @filter_type).order('name ASC').page(@page).per(@limit)
+        { json: { children: children_to_render,
+                  page: @page,
+                  total_pages: children_to_render.total_pages,
+                  total_records: children_to_render.length,
+                  limit: @limit },
+          except: [:medium_score],
+          status: :ok }
+      when Constants::USER_SORTINGS_ORDERS[:desc]
+        children_to_render = filter_profiles(children, @filter_type).order('name DESC').page(@page).per(@limit)
+        { json: { children: children_to_render,
```

Comment:
the response is almost the same, can you please conditionally calculate `children_to_render ` and then return 1 response (same for `sort_by_score `)